bottlerocket/agents/src/bin/vsphere-vm-resource-agent/vsphere_vm_provider.rs

```diff
@@ -136,15 +136,14 @@ impl Create for VMCreator {
 
         let vsphere_cluster = spec.configuration.cluster.clone();
         debug!("Decoding and writing out kubeconfig for vSphere cluster");
-        base64_decode_write_file(
-            &vsphere_cluster.kubeconfig_base64,
-            TEST_CLUSTER_KUBECONFIG_PATH,
-        )
-        .await
-        .context(
-            Resources::Clear,
-            "Failed to write out kubeconfig for vSphere cluster",
-        )?;
+        if let Some(kubeconfig_base64) = vsphere_cluster.kubeconfig_base64 {
```

Comment:
I probably shouldn't share the `VSphereK8sClusterInfo` struct definition between the vm resource agent and the cluster provider agent since the vm resource agent should never be invoked if the kubeconfig for the cluster is missing.

That should be enforced by the spec. By making the `kubeconfig_base64` field optional, we lose that validation.